server/data/data-jmap/src/main/scala/org/apache/james/jmap/memory/identity/MemoryCustomIdentityDAO.scala

```diff
@@ -0,0 +1,49 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.memory.identity
+
+import com.google.common.collect.{HashBasedTable, Table}
+import org.apache.james.core.Username
+import org.apache.james.jmap.api.identity.{CustomIdentityDAO, IdentityCreationRequest, IdentityNotFoundException, IdentityUpdate}
+import org.apache.james.jmap.api.model.{Identity, IdentityId}
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.jdk.CollectionConverters._
+
+class MemoryCustomIdentityDAO extends CustomIdentityDAO {
+  private val table: Table[Username, IdentityId, Identity] = HashBasedTable.create
+
+  override def save(user: Username, creationRequest: IdentityCreationRequest): Publisher[Identity] =
+    SMono.fromCallable(() => IdentityId.generate)
+      .map(creationRequest.asIdentity)
+      .doOnNext(identity => table.put(user, identity.id, identity))
```

Comment:
We should set default value for name, textSignature and htmlSignature in case they are empty (see https://jmap.io/spec-mail.html#identities)

Reply:
Also, adding tests for default values when empty input?

Reply:
Sure :+1:  missed this, good catch.